test/tests-using.js

```diff
@@ -0,0 +1,2485 @@
+          start: 26,
+          end: 38,
+          id: {
+            type: "Identifier",
+            start: 26,
+            end: 27,
+            name: "x"
+          },
+          init: {
+            type: "Identifier",
+            start: 30,
+            end: 38,
+            name: "resource"
+          }
+        }],
+        kind: "using"
+      }, {
+        type: "BreakStatement",
+        start: 40,
+        end: 46,
+        label: null
+      }]
+    }
+  }],
+  sourceType: "script"
+}, {"ecmaVersion": 17});
+
+// Using in labeled statement block
+test("label: { using x = resource; }", {
+  type: "Program",
+  start: 0,
+  end: 30,
+  body: [{
+    type: "LabeledStatement",
+    start: 0,
+    end: 30,
+    label: {
+      type: "Identifier",
+      start: 0,
+      end: 5,
+      name: "label"
+    },
+    body: {
+      type: "BlockStatement",
+      start: 7,
+      end: 30,
+      body: [{
+        type: "VariableDeclaration",
+        start: 9,
+        end: 28,
+        declarations: [{
+          type: "VariableDeclarator",
+          start: 15,
+          end: 27,
+          id: {
+            type: "Identifier",
+            start: 15,
+            end: 16,
+            name: "x"
+          },
+          init: {
+            type: "Identifier",
+            start: 19,
+            end: 27,
+            name: "resource"
+          }
+        }],
+        kind: "using"
+      }]
+    }
+  }],
+  sourceType: "script"
+}, {"ecmaVersion": 17});
+
+// Using with 'arguments' identifier
+test("{ using arguments = resource; }", {
+  type: "Program",
+  start: 0,
+  end: 31,
+  body: [{
+    type: "BlockStatement",
+    start: 0,
+    end: 31,
+    body: [{
+      type: "VariableDeclaration",
+      start: 2,
+      end: 29,
+      declarations: [{
+        type: "VariableDeclarator",
+        start: 8,
+        end: 28,
+        id: {
+          type: "Identifier",
+          start: 8,
+          end: 17,
+          name: "arguments"
+        },
+        init: {
+          type: "Identifier",
+          start: 20,
+          end: 28,
+          name: "resource"
+        }
+      }],
+      kind: "using"
+    }]
+  }],
+  sourceType: "script"
+}, {"ecmaVersion": 17});
+
+// Using with 'eval' identifier (should work, eval is not restricted in using declarations)
+test("{ using eval = resource; }", {
+  type: "Program",
+  start: 0,
+  end: 26,
+  body: [{
+    type: "BlockStatement",
+    start: 0,
+    end: 26,
+    body: [{
+      type: "VariableDeclaration",
+      start: 2,
+      end: 24,
+      declarations: [{
+        type: "VariableDeclarator",
+        start: 8,
+        end: 23,
+        id: {
+          type: "Identifier",
+          start: 8,
+          end: 12,
+          name: "eval"
+        },
+        init: {
+          type: "Identifier",
+          start: 15,
+          end: 23,
+          name: "resource"
+        }
+      }],
+      kind: "using"
+    }]
+  }],
+  sourceType: "script"
+}, {"ecmaVersion": 17}); 
```

Comment:
Could you please add the following test?
As I understand it can be parsed, but it is not an `unsing` declaration.

```js
for (using of x) {}
```

Reply:
fixed: https://github.com/acornjs/acorn/pull/1369/commits/c395a8bf581c01e86b8780637e4fe26cdf43a6be